### PR TITLE
added basic rpc framework

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description
- added basic RPC framework (will make cleaner in the next PR)
- got HELLO to work
- fixed issue with LRU cache that caused the same block to be processed multiple times

### CODE REVIEW

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/PeerManager.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          .thenApply(resp -> peer.getRemoteHelloMessage().complete(resp));
+    }
+  }
+
+  public CompletableFuture<?> connect(final Multiaddr peer, final NetworkImpl network) {
+    STDOUT.log(Level.INFO, "Connecting to " + peer);
+    return network
+        .connect(peer)
+        .whenComplete(
+            (conn, t) -> {
```
Comment (cemozerr) [inline networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/PeerManager.java:83]: I know this is tiny, but "t" feels weird for an error response variable. no?

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java [hunk 2]
```diff
@@ -184,8 +193,8 @@ private void onTick(Date date) {
           this.eventBus.post(new SlotEvent(nodeSlot));
           this.currentSlotGauge.set(nodeSlot.longValue());
           this.currentEpochGauge.set(compute_epoch_of_slot(nodeSlot).longValue());
-          STDOUT.log(Level.INFO, "******* Slot Event *******", ALogger.Color.WHITE);
-          STDOUT.log(Level.INFO, "Node slot:                             " + nodeSlot);
+          STDOUT.log(Level.DEBUG, "******* Slot Event *******", ALogger.Color.WHITE);
```
Comment (cemozerr) [inline services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java:196]: Why the switch to slot event?

Comment (cemozerr) [reply services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java:196]: switch to debug when printing*

**Review by cemozerr — APPROVED** [review]
Left a few comments. but looks good to me.

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/PeerManager.java (hunk 1 above)
Comment (jrhea) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/PeerManager.java:83]: agreed.  it was adapted from harmony.  i will modify it to match our style

Comment (jrhea) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/PeerManager.java:83]: fyi, t is short for Throwable so it isn't terrible, but i don't mind being more explicit with names

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java (hunk 2 above)
Comment (jrhea) [reply services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java:196]: yep, should be INFO.  I had several other printouts that i needed to be DEBUG and i did a search and replace in the file and missed this.  Good catch

**Pull request merged**